batch/batch/cloud/azure/resource_utils.py

```diff
@@ -197,8 +203,13 @@ def azure_requested_to_actual_storage_bytes(storage_bytes, allow_zero_storage):
         return None
     if allow_zero_storage and storage_bytes == 0:
         return storage_bytes
+
     # actual minimum storage size is 4 Gi on Azure, but keeping 10 to be consistent with gcp
-    return max(10 * 1024 ** 3, storage_bytes)
+    storage_bytes = max(10 * 1024 ** 3, storage_bytes)
+
+    # assuming Premium disks because this is used for computing user's external storage
+    disk = azure_disk_from_storage_in_gib('P', storage_bytes)
```

Comment:
We need to make sure the docs are crystal clear about what `.storage` means, because this suggests to me that it means "you will receive *at least* this much disk space, but you may pay for twice as much"

To be clear, I don't think that's a problem, in fact, I think powers of two of disk is a good idea. We might be able to do some kind of packing thing with powers of two disks.

Reply:
I agree that we need to make this clear when we have Azure specific docs. It's the same behavior in GCP right now where you get exactly what you requested.

Reply:
I guess it's not hard to add this to the docs right now. Will do so